### PR TITLE
Wrapper over all FDO functionalities in a single script - Adding Java…

### PR DESCRIPTION
**Pull request by NazarenoLopez**
Modified Java Version to 17 and the last docker-compose version 2.24.5

### CODE REVIEW

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+cp -r secrets service.env ../owner/
+cp -r secrets service.env ../aio/
+cp -r secrets ../db/
+}
+
+start_db() {
+cd $1/db
+if [ ! "$( docker container inspect -f '{{.State.Status}}' $container_name )" == "running" ]; then
+  sed -i 's+innodb_change_buffer_max_size = 25+#innodb_change_buffer_max_size = 25+g' custom/config-file.cnf
+  docker-compose up --build -d
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:190]: After executing the command, better to check if the container started properly and running using check


```suggestion
  docker-compose up --build -d
  if [ ! "$( docker container inspect -f '{{.State.Status}}' db_fdo-db_1 )" == "running" ]; then
     echo "Failed to start DB container" >&2
     exit 1
  fi
```

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+fi
+}
+
+start_aio() {
+cd $1/aio
+sed -i 's/  #- org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/  - org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/g' service.yml
+sed -i 's/  - org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/  #- org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/g' service.yml
+docker-compose up --build -d
+docker logs -f pri-fdo-aio
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:199]: This command follows log output of AIO container and might result in unwanted behavior, hence can be removed

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  sed -i 's+innodb_change_buffer_max_size = 25+#innodb_change_buffer_max_size = 25+g' custom/config-file.cnf
+  docker-compose up --build -d
+fi
+}
+
+start_aio() {
+cd $1/aio
+sed -i 's/  #- org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/  - org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/g' service.yml
+sed -i 's/  - org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/  #- org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/g' service.yml
+docker-compose up --build -d
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:198]: Similar comment to previous to check if container is started and running

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+docker logs -f pri-fdo-aio
+}
+
+start_mfg() {
+cd $1/manufacturer
+docker-compose up --build -d
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:204]: Similar comment to previous to check if container is started and running

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+}
+
+start_rv() {
+cd $1/rv
+sed -i 's/  #- org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/  - org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/g' service.yml
+sed -i 's/  - org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/  #- org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/g' service.yml
+docker-compose up --build -d
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:211]: Similar comment to previous to check if container is started and running

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+}
+
+start_owner() {
+cd $1/owner
+docker-compose up --build -d
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:216]: Similar comment to previous to check if container is started and running

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+   echo '127.0.0.1 host.docker.internal' | sudo tee -a /etc/hosts
+ fi
+
+ echo "Added host.docker.internal mappings"
+}
+
+aio_e2e_setup() {
+  add_docker_internal_mappings
+  install_dependencies
+  docker_cleaner
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:231]:
```suggestion
  fdo_docker_cleaner
```

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+#    chk $? 'linking docker-compose to /usr/bin'
+fi
+
+
+echo "Upgraded and Installed all FDO dependencies"
+}
+
+
+update_scripts() {
+cd $1
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:156]: Better to check if such path or directory that is provided as argument exists and then change directory to the same

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+echo "DNS.3 = host.docker.internal" >> web-server.conf
+echo "DNS.4 = host.docker.internal" >> web-server.conf
+echo "IP.1 = $ipaddress" >> web-server.conf
+echo "IP.2 = 127.0.0.1" >> web-server.conf
+echo "IP.3 = 172.17.0.1" >> web-server.conf
+
+}
+
+generate_keys() {
+cd $1
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:173]: Better to check if such path or directory that is provided as argument exists and then change directory to the same

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+chmod +r secrets/server-key.pem
+cp -r secrets service.env ../manufacturer/
+cp -r secrets service.env ../rv/
+cp -r secrets service.env ../owner/
+cp -r secrets service.env ../aio/
+cp -r secrets ../db/
+}
+
+start_db() {
+cd $1/db
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:187]: Better to check if such path or directory that is provided as argument exists and then change directory to the same

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+if [ ! "$( docker container inspect -f '{{.State.Status}}' $container_name )" == "running" ]; then
+  sed -i 's+innodb_change_buffer_max_size = 25+#innodb_change_buffer_max_size = 25+g' custom/config-file.cnf
+  docker-compose up --build -d
+fi
+}
+
+start_aio() {
+cd $1/aio
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:195]: Better to check if such path or directory that is provided as argument exists and then change directory to the same

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+  #  echo "Successfully updated rvinfo"
+  #fi
+#}
+
+
+
+# Main Execution script starts from here
+
+if [[ -z "${ipaddress}" ]]; then
+  ipaddress=`ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/'`
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:257]: Based on interface being used, this not might always return correct IP address , better to use following
```suggestion
  ipaddress=`ip route get 8.8.8.8 | grep -oP 'src \K[^ ]+'`
```

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+fi
+
+while getopts ":b:c:d:i:z" opt; do
```
Comment (Sai-Anudeep47) [inline component-samples/demo/scripts/fdo_toolkit.sh:260]: `:` can be removed for the options that has no arguments passed to them

Comment (NazarenoLopez) [reply component-samples/demo/scripts/fdo_toolkit.sh:313]: I don´t understand what you are trying to say. :-(

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+  fdo_docker_cleaner
+  download_and_unpack_binaries
+  update_scripts ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE/scripts
+  generate_keys ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE/scripts
+  start_db ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE
+  sleep 10
+  start_aio ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE
+}
+
+#update_rvinfo_aio() {
```
Comment (SSWAMIN1SSS) [inline component-samples/demo/scripts/fdo_toolkit.sh:293]: Can we remove the commented codes, if we are not using it?

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+start_aio() {
+if [ ! -e "$1" ]; then
+  echo "Error: $1 does not exist" >&2
+  exit 1
+fi
+cd $1/aio
+sed -i 's/  #- org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/  - org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/g' service.yml
+sed -i 's/  - org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/  #- org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/g' service.yml
+if [ ! "$( docker container inspect -f '{{.State.Status}}' pri-fdo-aio )" == "running" ]; then
+  sed -i 's+innodb_change_buffer_max_size = 25+#innodb_change_buffer_max_size = 25+g' custom/config-file.cnf
```
Comment (SSWAMIN1SSS) [inline component-samples/demo/scripts/fdo_toolkit.sh:220]: This sed operation is not required, as its not part of our current config-file.cnf.This was removed as part of DB Upgradation

File: component-samples/demo/scripts/fdo_toolkit.sh [hunk 16]
```diff
@@ -0,0 +1,335 @@
...
+# All methods will go here
+
+
+
+#################    Documentation   #############################
+#
+#
+#  Usage:
+#
+#  fdo_toolkit.sh
```
Comment (SSWAMIN1SSS) [inline component-samples/demo/scripts/fdo_toolkit.sh:14]: The script execution will work on Ubuntu-22/Ubuntu-20/Debian, It wont work on RHEL.
can it be updated in the scripts same as well?

File: component-samples/demo/scripts/fdo_toolkit.sh (hunk 13 above)
Comment (NazarenoLopez) [reply component-samples/demo/scripts/fdo_toolkit.sh:313]: Sai, please provide me a sample to the expected shell script like.

Comment (Sai-Anudeep47) [reply component-samples/demo/scripts/fdo_toolkit.sh:290]:
```suggestion
while getopts ":bcdiz" opt; do
```

**Pull request closed without merging**